opta/commands/generate_terraform.py

```diff
@@ -0,0 +1,304 @@
+import datetime
+import json
+import os
+import shutil
+import tempfile  # nosec
+from typing import Dict, Optional
+
+import click
+
+from opta.amplitude import amplitude_client
+from opta.commands.local_flag import _clean_folder, _clean_tf_folder
+from opta.constants import REGISTRY, TF_FILE_PATH, VERSION
+from opta.core.generator import gen, gen_opta_resource_tags
+from opta.layer import Layer
+from opta.pre_check import pre_check
+from opta.utils import check_opta_file_exists, dicts, logger
+from opta.utils.clickoptions import str_options
+from opta.utils.markdown import Code, Markdown, Text, Title1, Title2
+
+
+@click.command()
+@click.option(
+    "-c", "--config", default="opta.yaml", help="Opta config file", show_default=True
+)
+@click.option(
+    "-e",
+    "--env",
+    default=None,
+    help="The env to use when loading the config file",
+    show_default=True,
+)
+@click.option(
+    "-d",
+    "--directory",
+    default="./generated-terraform",
+    help="Directory for output",
+    show_default=True,
+)
+@click.option(
+    "--readme-format",
+    type=click.Choice(["none", "md", "html"], case_sensitive=False),
+    default="html",
+    help="Readme format",
+    show_default=True,
+)
+@click.option(
+    "--replace",
+    is_flag=True,
+    default=False,
+    help="""Replace the existing directory if it already exists.
+    Warning: If used, the existing files will be lost, including the terraform state files if local backend is used.""",
+    show_default=True,
+)
+@click.option(
+    "--auto-approve",
+    is_flag=True,
+    default=False,
+    help="Automatically approve confirmation message for deleting existing files.",
+)
+@click.pass_context
+def generate_terraform(
+    ctx: click.Context,
+    config: str,
+    env: Optional[str],
+    directory: str,
+    readme_format: str,
+    replace: bool,
+    auto_approve: bool,
+) -> None:
+    """Generate Terraform language files
+
+    Examples:
+
+    opta generate-terraform -c my-config.yaml
+
+    opta generate-terraform -c my-config.yaml -d ./tf
+    """
+
+    if directory.strip() == "":
+        raise click.UsageError("--directory can't be empty")
```

Comment:
We may want to add a check that the directory is not already populated with an older run of generate terraform; otherwise TF files may get mixed up...(e.g. I generate terraform for a staging env, then for a product env, and then all the files are mixed up in this directory.)

Reply:
good point, I added a confirmation check for this, see `The output directory will be updated.\n Do you approve?` below

Reply:
thanks